src/Products/PlonePAS/tests/test_membershiptool.py

```diff
@@ -700,6 +704,26 @@ def testGetBadMembers(self):
                                              TEST_USER_ID)
         self.assertEqual(self.membership.getBadMembers(), [])
 
+    def test_credentials_updated_event(self):
+        events_fired = []
+
+        @adapter(ICredentialsUpdatedEvent)
+        def got_credentials_updated_event(event):
+            events_fired.append(event)
+
+        gsm = getGlobalSiteManager()
+        gsm.registerHandler(got_credentials_updated_event)
+
+        self.assertTrue(self.membership.testCurrentPassword('secret'))
+        self.assertFalse(self.membership.testCurrentPassword('whoknows'))
+        login(self.portal, TEST_USER_NAME)  # Back to normal
+        self.membership.setPassword('guessagain')
+        self.assertEqual(len(events_fired), 1)
+        self.assertEqual(events_fired[0].principal.getId(), TEST_USER_ID)
+        self.assertEqual(events_fired[0].password, 'guessagain')
+
+        gsm.unregisterHandler(got_credentials_updated_event)
```

Comment:
Do we need to unregister? Anyway, it does not hurt, but the test-layer should take care of it.

Reply:
Done!